File: day01/cpp/day01.cpp
#include <iostream>
#include <string>
#include <fstream> 
using namespace std; 
 
int getAnswer(int mass){
	if(mass<0)
		return 0;
	int fuel = (mass/3)-2;
	return fuel + getAnswer(fuel);
} 

int main() 
{ 
    fstream file; 
    string word, filename;
    filename = "input.txt"; 

    // PART 1
    file.open(filename.c_str());  
    int answer = 0;
    while (file >> word) 
        answer += (stoi(word)/3)-2;
    cout<<answer<<endl; 

    // PART 2
    int answer2=0;
    file.clear();
	file.seekg(0, ios::beg);
    while(file>>word)
    	answer2 += getAnswer(stoi(word));
    cout<<answer2<<endl;
    file.close();
   return 0; 
 }

File: day01/day01.py
import os
import math
from functools import reduce

filepath = os.path.join('.', 'input.txt')
answer = 0

with open(filepath, 'r') as file_handler:
    values = file_handler.read()
    lines = values.split('\n')
    for line in lines:
        current = int(line)
        answer = answer + math.floor(current / 3) - 2
print(answer)

with open(filepath, 'r') as file_handler:
    values = file_handler.read()
    lines = values.splitlines()


    def getFuel(mass):
        fuel = math.floor(mass / 3) - 2
        return fuel + getFuel(fuel) if fuel > 0 else 0


    answer = reduce(lambda accum, element: accum + getFuel(int(element)), lines, 0)
    print(answer)
